### DISCUSSION TITLE
Is D_from_matrix differentiable?

### DISCUSSION THREAD

**Discussion by leolou0512** [e3nn/e3nn#382] [Q&A]
Hi,

I am trying to use e3nn to predict dielectric tensors of materials. I want to use the anisotropy ratio (defined as maximum eigenvalue divided by the smallest eigenvalue) as part of my loss. The problem is that I converted my 3x3 cartesian dielectric tensor matrix into irreps (1x0e+1x2e) for e3nn, and to evaluate the anisotropy ratio I have to convert irreps back to cartesian tensors using to_cartesian(). My attempts have not worked so far, so I'm wondering if back propagation can go through to_cartesian(), and thus learnable at all? thanks.

**Comment by mariogeiger** [comment]
Yes, the gradients can go through `to_cartesian ()`
